### PR TITLE
S cat conflation

### PR DESCRIPTION
**Pull request by maxTarlov**
Re: Issue #137 
This is the back-end for the tree trimming functionality. Adds 4 global functions:
* copyTree(): makes a copy of a tree's values in a separate reference so that the function other don't delete material from the input
* trimSilentTerminals(): removes silent terminals from the tree
* trimDeadEndNodes(): removes non-x0 terminal nodes that might have been created by the previous function if all of their children were silent
* trimRedundantNodes(): removes needless recursion such as [[arbitrary terminals]]. This might be in the input, but it might also have been made by the previous two functions.

I also implemented the tree trimming functionality into matchSP as an option that defaults to true. Should I do this for align as well?

### CODE REVIEW

File: constraints/match.js [hunk 1]
```diff
@@ -112,10 +112,15 @@ function matchPS(sTree, pParent, pCat, options)
 *	minProsody: If true, the prosodic match needs to be minimal. Passed to hasMatch.
 *	nonMaxProsody: If true, the prosodic match must be non-maximal. Passed to hasMatch.
 *	nonMinProsody: If true, the prosodic match must be non-minimal. Passed to hasMatch.
+* trimTrees: Default true, trims trees of silent nodes, non-x0 terminals and redundant nodes, e.g. [[some terminals]]
```
Comment (jbellik) [inline constraints/match.js]: I think this should default to false, actually. Also we should make sure that the tableau displays whether trimming has taken place or not.

**Comment by jbellik** [comment]
My other concern is what this function is doing to ptrees if you run MatchPS...

File: constraints/match.js (hunk 1 above)
Comment (maxTarlov) [reply constraints/match.js]: ok, updated in commit 6ddcac2

**Comment by maxTarlov** [comment]
new changes:
* trimTrees defaults to false
* matchPS deals with trimTrees so matchSP doesn't trim pTrees
* added the parenthesized trimmed tree to the tableaux when constraints use the trimTrees option

**Comment by maxTarlov** [comment]
trimming of trees is now only an option in makeTableau

File: test/treeTrimming.html [hunk 2]
```diff
@@ -0,0 +1,160 @@
+<!--Max Tarlov testing tree trimming 10/23/19-->
```
Comment (jbellik) [inline test/treeTrimming.html]: This has obviously been updated since October.

**Comment by jbellik** [comment]
@nkalivoda I'm going to merge this into master unless you have any objections.

**Pull request merged**